### PR TITLE
Standardize process exit codes

### PR DESCRIPTION
**Pull request by brandenrodgers**
## Description and Context

Following up this PR here: https://github.com/HubSpot/hubspot-cli/pull/565

We would like to have three different exit codes:
- `0`: A successful run
- `1`: There was a config problem or an internal error
- `2`: There are warnings or validation issues

I'm adding some utils for us to use around the repo to make sure that we're using the exit codes consistently. There are a lot of places that will need to be updated, so I haven't done that yet. I'm showing three different patterns that we could use. I don't feel super strongly about any of them, but I figured I'd check in with everyone before picking one of them.

Here's some more context. This is how the [ESLint CLI](https://eslint.org/docs/user-guide/command-line-interface) handles their error codes.

## TODO

- [x] Pick a pattern, and update all the commands to use it
## Who to Notify

cc/ @gcorne

### CODE REVIEW

**Comment by gcorne** [comment]
My preference is the enum with process.exit() because I think it makes it clearer that execution will stop at that point.

**Comment by anthmatic** [comment]
> My preference is the enum with process.exit() because I think it makes it clearer that execution will stop at that point. 

I second this

**Review by anthmatic — APPROVED** [review]
Looks good

File: packages/cli/lib/exitCodes.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+ * 0: Successful run
+ * 1: Config problem or internal error
+ * 2: Warnings or validation issues
+ */
+const EXIT_CODES = {
```
Comment (miketalley) [inline packages/cli/lib/exitCodes.js:6]: Would it make sense to put this under `/cli/lib/enums/exitCodes.js`?

Comment (brandenrodgers) [reply packages/cli/lib/exitCodes.js:6]: I can do that 👍

**Pull request merged**